src/rgw/rgw_crypt.cc

```diff
@@ -0,0 +1,1338 @@
+// -*- mode:C; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/**
+ * Crypto filters for Put/Post/Get operations.
+ */
+#include <rgw/rgw_op.h>
+#include <rgw/rgw_crypt.h>
+#include <auth/Crypto.h>
+#include <rgw/rgw_b64.h>
+#include <rgw/rgw_rest_s3.h>
+#include "include/assert.h"
+#include <boost/utility/string_ref.hpp>
+#include <rgw/rgw_keystone.h>
+#include "include/str_map.h"
+#include "crypto/crypto_accel.h"
+#include "crypto/crypto_plugin.h"
+#ifdef USE_NSS
+# include <nspr.h>
+# include <nss.h>
+# include <pk11pub.h>
+#endif
+
+#ifdef USE_CRYPTOPP
+#include <crypto++/cryptlib.h>
+#include <crypto++/modes.h>
+#include <crypto++/aes.h>
```

Comment:
i wasn't able to build with `USE_CRYPTOPP` because it can't find these headers under `<crypto++/`. this was on fedora with the cryptopp-devel package. elsewhere in the ceph tree, we're including cryptopp headers with `<cryptopp/` - making that change here allows me to build successfully, and the encryption s3tests all pass

i added a commit with this fix to the https://github.com/ceph/ceph-ci/commits/wip-rgw-encryption branch